src/ucp/proto/proto_common.c

```diff
@@ -320,7 +320,8 @@ static ucp_lane_index_t ucp_proto_common_find_lanes_internal(
                 continue;
             }
 
-            if (!(md_attr->cap.access_mem_types &
+            if (!(md_attr->cap.flags & UCT_MD_FLAG_NEED_RKEY) &&
+                !(md_attr->cap.access_mem_types &
```

Comment:
pls add code comment here to explain and squash (without rebase)

Reply:
you mean add fixes  before merge commit?